qa/acceptance-tests/src/test/java/io/camunda/it/auth/UserTaskRestrictionsIT.java

```diff
@@ -0,0 +1,246 @@
+    awaitUserTaskBeingAvailable(adminClient, processInstanceKeyCandidateGroups, 2);
+  }
+
+  @Test
+  public void searchShouldReturnCandidateUserTasks() throws JsonProcessingException {
+    // given
+    try (final var tasklistUser1RestClient =
+            STANDALONE_CAMUNDA
+                .newTasklistClient()
+                .withAuthentication(USER1_USER.username(), USER1_USER.password());
+        final var tasklistUser2RestClient =
+            STANDALONE_CAMUNDA
+                .newTasklistClient()
+                .withAuthentication(USER2_USER.username(), USER2_USER.password()); ) {
+
+      // when
+      // searching for tasks available to user1 and user2 (no process instance key provided)
+      final HttpResponse<String> user1Response = tasklistUser1RestClient.searchTasks(null);
+      final HttpResponse<String> user2Response = tasklistUser2RestClient.searchTasks(null);
+
+      // then
+      assertThat(user1Response).isNotNull();
+      assertThat(user2Response).isNotNull();
+      assertThat(user1Response.statusCode()).isEqualTo(200);
+      assertThat(user2Response.statusCode()).isEqualTo(200);
+
+      final TaskSearchResponse[] tasksUser1 =
+          TestRestTasklistClient.OBJECT_MAPPER.readValue(
+              user1Response.body(), TaskSearchResponse[].class);
+      assertThat(tasksUser1.length).isEqualTo(1);
+      assertThat(tasksUser1[0].getCandidateUsers()).containsExactly(USER1);
+      final var tasksUser2 =
+          TestRestTasklistClient.OBJECT_MAPPER.readValue(
+              user2Response.body(), TaskSearchResponse[].class);
+      assertThat(tasksUser2).hasSize(1);
+      assertThat(tasksUser2[0].getCandidateUsers()).containsExactly(USER2);
+    }
+  }
+
+  @Test
+  public void searchShouldReturnCandidateGroupTasks() throws JsonProcessingException {
+    // given
+    try (final var tasklistUser3RestClient =
+            STANDALONE_CAMUNDA
+                .newTasklistClient()
+                .withAuthentication(USER3_USER.username(), USER3_USER.password());
+        final var tasklistUser4RestClient =
+            STANDALONE_CAMUNDA
+                .newTasklistClient()
+                .withAuthentication(USER4_USER.username(), USER4_USER.password()); ) {
+
+      // when
+      // searching for tasks available to user1 and user2 (no process instance key provided)
+      final HttpResponse<String> user3Response = tasklistUser3RestClient.searchTasks(null);
+      final HttpResponse<String> user4Response = tasklistUser4RestClient.searchTasks(null);
+
+      // then
+      assertThat(user3Response).isNotNull();
+      assertThat(user4Response).isNotNull();
+      assertThat(user3Response.statusCode()).isEqualTo(200);
+      assertThat(user4Response.statusCode()).isEqualTo(200);
+
+      final TaskSearchResponse[] tasksUser3 =
+          TestRestTasklistClient.OBJECT_MAPPER.readValue(
+              user3Response.body(), TaskSearchResponse[].class);
+      assertThat(tasksUser3.length).isEqualTo(1);
```

Comment:
nit: for consistency and readability
```suggestion
      assertThat(tasksUser3).hasSize(1);
```

Reply:
It looks like this wasn't applied